### DISCUSSION TITLE
powershell on RHEL 9 based (Rocky Linux 9.4) authentication with Windows 2016 server

### DISCUSSION THREAD

**Discussion by grydan** [PowerShell/PowerShell#24092] [Q&A]
Hi,
I need to move my old application script .ps1 from Centos 7 to Rocky Linux 9.4 (RHEL 9.x based).
I have installed powershell latest version at now **PowerShell 7.4.4** and I have a simple script **test.ps1** with this code:

```
# Convert password in SecureString
$password = ConvertTo-SecureString "MYPWD_ADMIN_WIN2016" -AsPlainText -Force

# Create  PSCredential with username and pwd
$credential = New-Object System.Management.Automation.PSCredential("MYDOMAIN\administrator", $password)

# Start a session
$session = New-PSSession -ComputerName IP_WIN2016 -Credential $credential -Authentication Negotiate
```
this is the output

> Line |
>    8 |  $session = New-PSSession -ComputerName **IP_WIN2016** -Credential $cred …
>      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>      | This parameter set requires WSMan, and no supported WSMan client library was found. WSMan is either not installed or unavailable for this system.
> 

In Centos 7 I remember that with a simple this will be solved

**`yum install gss-ntlmssp`**

But in RHEL 9 this package was removed.

I read online that I need **omi** ; I have installed but this seem don't solve...

Have you any suggestion, solution for this simple problem?

**Comment by rhubarb-geek-nz** [comment]
[WSMan remoting isn't supported on non-Windows platforms](https://learn.microsoft.com/en-us/powershell/scripting/security/remoting/wsman-remoting-in-powershell?view=powershell-7.4#wsman-remoting-isnt-supported-on-non-windows-platforms)

> On non-Windows, WSMan relied on the [Open Management Infrastructure (OMI)](https://github.com/Microsoft/omi) project, which no longer supports PowerShell remoting. The OMI WSMan client is dependent on OpenSSL 1.0. Most Linux distributions have moved to OpenSSL 2.0, which isn't backward-compatible. At this time, there is no supported distribution that has the dependencies needed for the OMI WSMan client to work.

**Comment by kasini3000** [comment]
I have to sadly tell you that it is difficult to achieve various distributions.
But it can be implemented using various Docker images, such as:
```
docker pull malthaeldai/powershell-ntlm
```
Perhaps there are updated versions of the image, please let me know
